### PR TITLE
Enable more shorthand strategy names in the Fabric CLI

### LINKED ISSUE #16479: Expose available strategies from registry in Fabric CLI
### Description & Motivation

The Fabric CLI has the following options for strategy:

```
lightning run model --help

  --strategy [ddp|dp|deepspeed]   Strategy for how to run across multiple
                                  devices.
```

But special keywords like `deepspeed_stage_3` are not exposed. Why?

The initial implementation was restrictive and minimal with intention. The CLI can't support strategy options like "ddp-spawn", because it assumes the launcher is a script launcher.

### Pitch

Allow all entries from `StrategyRegistery.available_strategies()`, but filter out the ones that have the `MultiProcessingLauncher` (ddp-spawn, ddp-fork, etc.).

### Alternatives

1) Hardcode the subset of options without `StrategyRegistery.available_strategies()`
2) Keep as is

### Additional context

Raised by @justusschock 

cc @borda @carmocca @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #16479

For example, this is now possible:

```
lightning run model --strategy=deepspeed_stage_3 train.py
```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: tests/tests_fabric/test_cli.py [hunk 1]
```diff
@@ -54,7 +56,7 @@ def test_cli_env_vars_accelerator(_, accelerator, monkeypatch, fake_script):
     assert os.environ["LT_ACCELERATOR"] == accelerator
 
 
-@pytest.mark.parametrize("strategy", ["dp", "ddp", "deepspeed"])
+@pytest.mark.parametrize("strategy", ["dp", "ddp", "deepspeed", "deepspeed_stage_3"])
```
Comment (justusschock) [inline tests/tests_fabric/test_cli.py:59]:
```suggestion
@pytest.mark.parametrize("strategy", _get_supported_strategies())
```

File: tests/tests_fabric/test_cli.py [hunk 2]
```diff
@@ -65,6 +67,15 @@ def test_cli_env_vars_strategy(_, strategy, monkeypatch, fake_script):
     assert os.environ["LT_STRATEGY"] == strategy
```
Comment (justusschock) [inline tests/tests_fabric/test_cli.py:68]: Can we also add a test that just tests the values of `_get_supported_strategies`? Either by length or by comparing the actual values?

**Review by justusschock — APPROVED** [review]

File: tests/tests_fabric/test_cli.py (hunk 2 above)
Comment (awaelchli) [reply tests/tests_fabric/test_cli.py:68]: Is this the same request as your suggestion above. If not, I don't understand what you meant.

Comment (justusschock) [reply tests/tests_fabric/test_cli.py:68]: the request above is to test this for every strategy. Here I want something like

```
assert len(_get_supported_strategies()) == 42
assert "deepspeed" in _get_supported_strategies()
...
```

**Review by carmocca — APPROVED** [review]

File: tests/tests_fabric/test_cli.py (hunk 2 above)
Comment (awaelchli) [reply tests/tests_fabric/test_cli.py:68]: Got it. added

**Pull request merged**